app/ui-react/syndesis/src/modules/integrations/components/editor/EditorStepsWithToolbar.tsx

```diff
@@ -0,0 +1,126 @@
+import { Connection, ConnectionOverview } from '@syndesis/models';
+import {
+  ConnectionsListView,
+  IActiveFilter,
+  IConnectionsListViewProps,
+  IFilterType,
+  ISortType,
+} from '@syndesis/ui';
+import { WithListViewToolbarHelpers } from '@syndesis/utils';
+import * as React from 'react';
+import { Translation } from 'react-i18next';
+import i18n from '../../../../i18n';
+import resolvers from '../../../connections/resolvers';
+import { EditorSteps, IEditorStepsProps } from './EditorSteps';
+import { IUIIntegrationStep } from './interfaces';
+
+function getFilteredAndSortedConnections(
+  steps: IUIIntegrationStep[],
+  activeFilters: IActiveFilter[],
+  currentSortType: ISortType,
+  isSortAscending: boolean
+) {
+  let filteredAndSortedConnections = steps;
+  activeFilters.forEach((filter: IActiveFilter) => {
+    const valueToLower = filter.value.toLowerCase();
+    filteredAndSortedConnections = filteredAndSortedConnections.filter(
+      (c: Connection) => c.name.toLowerCase().includes(valueToLower)
+    );
+  });
+
+  filteredAndSortedConnections = filteredAndSortedConnections.sort(
+    (miA, miB) => {
+      const left = isSortAscending ? miA : miB;
+      const right = isSortAscending ? miB : miA;
+      return left.name.localeCompare(right.name);
+    }
+  );
+
+  return filteredAndSortedConnections;
+}
+
+const filterByName = {
+  filterType: 'text',
+  id: 'name',
+  placeholder: i18n.t('shared:filterByNamePlaceholder'),
+  title: i18n.t('shared:Name'),
+} as IFilterType;
+
+const filterTypes = [filterByName];
+
+const sortByName = {
+  id: 'name',
+  isNumeric: false,
+  title: i18n.t('shared:Name'),
+} as ISortType;
+
+const sortTypes: ISortType[] = [sortByName];
+
+export interface IEditorStepsWithToolbarProps
+  extends IEditorStepsProps,
+    Pick<IConnectionsListViewProps, 'createConnectionButtonStyle'> {
+  children?: any;
+}
+
+export class EditorStepsWithToolbar extends React.Component<
+  IEditorStepsWithToolbarProps
+> {
+  public static defaultProps = {
+    includeHidden: false,
+  };
+
+  public render() {
+    return (
+      <Translation ns={['connections', 'shared']}>
+        {t => (
+          <WithListViewToolbarHelpers
+            defaultFilterType={filterByName}
+            defaultSortType={sortByName}
+          >
+            {helpers => {
+              const connectionString = 'connection';
+
+              const filteredAndSortedConnections = getFilteredAndSortedConnections(
+                this.props.connections as IUIIntegrationStep[],
+                helpers.activeFilters,
+                helpers.currentSortType,
+                helpers.isSortAscending
+              ).map(connection => {
+                return connection[connectionString];
```

Comment:
Not all items have a connection property I think? I’ll have to check this tomorrow against a BE but I think we might be ending up not showing all the steps

Reply:
Yeah, only `endpoint` kinds would have a `connection` property.  But we could be misled here if  it's just because `connection` needs to be renamed to something like `editorStep`.

Reply:
Renaming to `editorStep[connectionString]`. I can do a check for `uiStepKind === 'editor-step'`, but at the moment only one of all the listed items isn't `endpoint`, and that's `api-provider`. Also, from what I'm seeing, even `api-provider` has a `connection` property. This is how the list appears for me prior to `.map()`: https://gist.github.com/kahboom/d58532286c0f94da46ef17e0c477ba73

In the raw version, grep `api-provider` as an example.

Reply:
`editor-step` [isn't a valid value](https://github.com/syndesisio/syndesis-react/blob/4cbbdfbe57f0e08ea133138551a7752dccacf737/app/ui-react/syndesis/src/modules/integrations/components/editor/interfaces.ts#L34) for uiStepKind.

Again, we are not dealing with connections, but with a mixed bag of different things: connections, stepkinds and extensions. All those types have been normalized as `IUIIntegrationStep` objects. You'll have a valid `title` - to be used for the sorting an filtering - and for some objects a valid `connector` object. All will have some `metadata` set.
All this info needs to be properly handled to show - or not - the tech preview/configuration required badges.

Reply:
I'm aware that we are not dealing with connections, but what I'm getting for the data is an array of objects each with a parent `connection` property.
```
[
  { connection: { configuredProperties: { .. }, connector: { .. }  } },
  { connection: { .. } },
  { connection: { .. } },
  and so on..
]
```
I can either `.map()` it or similar, or I can write some type of interface like
```
export interface IUIEditorStep extends IUIIntegrationStep {
  connection: IUIStep
}
```
Neither is ideal, but if you have a different approach in mind please share what you are looking for exactly.

The original purpose of the PR was to fix the Tech Preview feature that was previously working, if we want to roll out a new toolbar for connections and are changing the object, that's getting kinda out of the scope of this PR.

Reply:
Can you re-read what @riccardo-forina said?  You can't rely on `connection` being set for this particular collection.  Much like the old angular app tried to re-use the "Connection List" page we actually kinda started in the same path here in the React app, as I recall when this page was originally written I had just created the combined step/connection/extension step in the angular app.  So this component that you're writing has to deal with that same combination.  At the very least, these objects are normalized into an IUIStep, so you should only need to look at the `metadata` value for each thing in the collection, not the `connection` object directly.  Also, use the `getMetadataValue<string>`, helper just in the off chance an item in the collection doesn't have it set.  I haven't taken a close look to see if you're using that here.